### PR TITLE
Fix various bugs where checks were missing for the presence of data/p…

### LINKED ISSUE #15: Non-existent auth causes traceback
https://github.com/candlepin/rho/issues/9

If we try to add/edit a profile and give it an auth that doesn't exist, we
get an exception, and then the config is busted.
```
Profile f11-bogus-multiauth-multiport edited
[adrian@alikins rho]$ rho dumpconfig
Traceback (most recent call last):
File "/home/adrian/src/rho/bin/rho", line 22, in
CLI().main()
File "/home/adrian/src/rho/src/rho/cli.py", line 95, in main
cmd.main()
File "/home/adrian/src/rho/src/rho/clicommands.py", line 133, in main
self.config = self._read_config(self.options.config, self.passphrase)
File "/home/adrian/src/rho/src/rho/clicommands.py", line 98, in _read_config
return config.ConfigBuilder().build_config(confstr)
File "/home/adrian/src/rho/src/rho/config.py", line 289, in build_config
config = Config(auths=creds, profiles=profiles)
File "/home/adrian/src/rho/src/rho/config.py", line 93, in init
self.add_profile(profile)
File "/home/adrian/src/rho/src/rho/config.py", line 134, in add_profile
c)
rho.config.ConfigError: No such credentials: rsa_key
```

### PR DESCRIPTION
**Pull request by chambridge**
…rofiles or data/credentials. Closes #15.

Fix various bugs while playing with rho command line vagrant functional testing.
```
~/Code/rho$ rho profile clear --all
All network profiles removed
~/Code/rho$ rho auth clear --all
All authorization credentials removed
#Test profile creation without auth existing
~/Code/rho$ rho profile add --name=vagrant-local --hosts 127.0.0.1:2222 --auth rho-dev_auth
No credentials exist yet.
#create auth
~/Code/rho$ rho auth add --name=rho-dev_auth --sshkeyfile=./.vagrant/machines/rho-dev/virtualbox/private_key --username=vagrant
~/Code/rho$ rho profile add --name=vagrant-local --hosts 127.0.0.1:2222 --auth rho-dev_auth
#Edit profile with non-existent auth
~/Code/rho$ rho profile edit --name=vagrant-local --hosts 127.0.0.1:2222 --auth test_1_auth
Auths do not exist.
```

### CODE REVIEW

File: rho/clicommands.py [hunk 1]
```diff
@@ -266,7 +266,8 @@ def _create_ping_inventory(profile_ranges, profile_auth_list, forks):
 
         out = out.split('\n')
 
-        for line in enumerate(out):
+        # pylint: disable=consider-using-enumerate
```
Comment (noahl) [inline rho/clicommands.py]: Why did you remove the `enumerate`? I'm not objecting, I just don't understand the reasoning yet.

File: rho/clicommands.py [hunk 2]
```diff
@@ -810,6 +822,10 @@ def _do_command(self):
 
         # removes all inventories ever.
         elif self.options.all:
+            if not os.path.isfile('data/profiles'):
+                print(_("All network profiles removed"))
+                sys.exit(0)
```
Comment (noahl) [inline rho/clicommands.py]: Why `sys.exit(0)` here? It doesn't match the control flow before the change, and it seems to mean that we have to run this command last or else rho will die prematurely.

File: test/test_clicommand.py [hunk 3]
```diff
@@ -132,3 +133,23 @@ def test_bad_range_options(self):
                            "profilename", "hosts",
                            "a:d:b:s", "--auths",
                            "auth_1", "auth2"])
+
+    def test_a_auth_add(self):
```
Comment (noahl) [inline test/test_clicommand.py:137]: Thank you for adding unit tests! What do `_a_` and `_z_` mean?

File: rho/clicommands.py (hunk 1 above)
Comment (chambridge) [reply rho/clicommands.py]: The original code used `range(len(out))` before my lint changes. I encountered an error when running the actual `rho` command with this so I backed out the change for the time being.

Comment (noahl) [reply rho/clicommands.py]: I see. I think you want `for line, _ in enumerate(out):`. `enumerate` returns a pair and not an integer, so using its output as an index into a list should have produced an error.

File: test/test_clicommand.py (hunk 3 above)
Comment (chambridge) [reply test/test_clicommand.py:137]: I tried using http://pytest-ordering.readthedocs.io/en/develop/#other-markers to order the tests, but that didn't seem to work for me. It runs tests in alphabetical order currently. These prefixes control when the test are getting run. I essentially run the add auth and add profile commands before all others and then clear all auth and profile as the last tests.

File: rho/clicommands.py (hunk 2 above)
Comment (chambridge) [reply rho/clicommands.py]: I changed this to add an else clause instead of the `sys.exit(0)`.

File: test/test_clicommand.py (hunk 3 above)
Comment (noahl) [reply test/test_clicommand.py:137]: I think it would be better to combine the add and clear steps into single tests rather than depending on the order the tests are run in. But up to you.

Comment (chambridge) [reply test/test_clicommand.py:137]: I'll open an issue regarding this. Currently the existing test cases depended on the fact that the files `data/credentials` and `data/profiles` are checked into git, which they shouldn't be.

Comment (chambridge) [reply test/test_clicommand.py:137]: https://github.com/quipucords/rho/issues/33

Comment (noahl) [reply test/test_clicommand.py:137]: Fine with me.

**Comment by noahl** [comment]
LGTM
